sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/core/PropertiesImpl.kt

```diff
@@ -12,7 +12,7 @@ data class SqlDelightDatabasePropertiesImpl(
   override val compilationUnits: List<SqlDelightCompilationUnitImpl>,
   override val className: String,
   override val dependencies: List<SqlDelightDatabaseNameImpl>,
-  override val dialectPreset: DialectPreset = DialectPreset.SQLITE_3_18,
+  override val dialectPresetName: String = DialectPreset.SQLITE_3_18.name,
```

Comment:
the `BuildAction` subclass gets serialized and run in a process whose classpath is both not the IDE classpath or the target gradle build's classpath. Configuration maybe? Anyways, `DialectPreset` wasn't available on that classpath so I switched to a string